### PR TITLE
[WIP] Merge develop to master

### PR DESCRIPTION
**Pull request by kkellerlbl**
Merge develop to master (no staging branch) for production deploy.

### CODE REVIEW

File: server_scripts/build_server_control_scripts.py [hunk 1]
```diff
@@ -62,26 +62,25 @@ def getConfig(param, cfg, cfile):
...
+        .format(maxmem,port,serviceDir,serviceDir))
 
 with open(os.path.join(serviceDir, 'stop_service'), 'w') as ss:
     ss.write('export JAVA_HOME={}\n'.format(javaHome))
     ss.write('export PATH=$JAVA_HOME/bin:$PATH\n')
     ss.write('export CLASSPATH=\n')
-    ss.write(('{}/glassfish_administer_service.py --admin {} ' +
-        '--domain {} --domain-dir {}/glassfish_domain --port {}\n')
-        .format(serviceDir, asadmin, serviceDomain, serviceDir, port))
+    ss.write('killall java\n')
```
Comment (MrCreosote) [inline server_scripts/build_server_control_scripts.py:83]: this will kill all java processes on the system, including, say, eclipse. That's not ok.

Comment (kkellerlbl) [inline server_scripts/build_server_control_scripts.py:83]: It's probably fine to remove it, though some people would say Eclipse deserves it.

It's too bad this didn't get caught when the code was committed, but it's a trivial enough change that making it now shouldn't be too much trouble.

Comment (MrCreosote) [inline server_scripts/build_server_control_scripts.py:83]: If you do that the shutdown script won't work.

Comment (kkellerlbl) [inline server_scripts/build_server_control_scripts.py:83]: Yeah, I just realized that.  We need pidfile support.  I may be able to add it tomorrow but no promises.

Comment (MrCreosote) [inline server_scripts/build_server_control_scripts.py:83]: What's the consequences if this doesn't get merged for (say) a couple months? Do we need to switch to Jetty right this second?

BTW, shock_service uses daemonize for this sort of thing which seems to work well.

**Comment by MrCreosote** [comment]
Actually, another question - I assume this change has been well tested? Since I'm responsible for the code base I suppose I should look at any test results showing the new server tech works as well or better than the old one.

**Comment by kkellerlbl** [comment]
It's been deployed on -ci for months.  The reason for the promotion is to deploy to production what's currently deployed in -next/-ci.  I do not have a strong preference to deploying this change or not (since it's just jetty instead of glassfish; I do have a slight preference to merging), but maybe @scanon has one.

**Comment by MrCreosote** [comment]
How much traffic does ci get compared to prod?

**Comment by kkellerlbl** [comment]
I don't know numbers but -ci has gotten a ton of use by us during the past few weeks of development.  At this point we may even have more developers than real production users.

**Comment by kkellerlbl** [comment]
Right now the glassfish deploy is still running on -next (because there's no staging branch for this repo).  So it wouldn't hurt my feelings to skip this merge for now.

**Comment by MrCreosote** [comment]
Works for me; we'll just have to find some other way to hurt your feelings.

**Comment by MrCreosote** [comment]
Twerp.

**Comment by kkellerlbl** [comment]
I'm closing this but I will remember your insolence.

**Comment by msneddon** [comment]
FYI- the killall issue affects all the services that were ported to Jetty, so it's in the NMS and user profile services now too, possibly others.  I flagged this early on in jira before the PR (https://atlassian.kbase.us/browse/KBASE-2127) but the developer who was working on the jetty transition left kbase before this was fixed, and no one followed up.  Any volunteers who want to follow up now?

**Comment by kkellerlbl** [comment]
I can take it, but probably not till after the DOE meeting.  (I probably should have back then but was too busy.  :( )

**Comment by MrCreosote** [comment]
Another note - Jetty UJS is _not_ running on CI. UJS is still glassfish everywhere.

**Comment by MrCreosote** [comment]
Made a dev-jetty branch at the head of develop and deleted develop.

**Pull request closed without merging**